src/lib.rs

```diff
@@ -192,59 +312,136 @@ impl<'object, Endian> DebugInfo<'object, Endian>
                 }
             }
         }
-        None
+
+        loc.map(move |(f, l)| (f, l, func.map(|u| u.0.name.to_string_lossy())))
     }
 }
 
-// TODO: most of this should be moved to the main library.
-struct Unit<'input> {
-    address_size: u8,
-    ranges: Vec<gimli::Range>,
-    line_offset: gimli::DebugLineOffset,
-    comp_dir: Option<&'input std::ffi::CStr>,
-    comp_name: Option<&'input std::ffi::CStr>,
+enum Target<'input> {
+    Line(LineUnit<'input>),
+    Function(FunctionUnit<'input>),
```

Comment:
Is there any reason these need to be in the same enum, instead of having separate `Vec<LineUnit>` and `Vec<FunctionUnit>`? There's probably a small memory saving from having them separate.